src/main/java/com/ldtteam/structurize/management/StructureName.java

```diff
@@ -106,13 +109,18 @@ else if (name.startsWith(Structures.SCHEMATICS_CACHE + Structures.SCHEMATICS_SEP
         {
             hut = schematic.split("\\d+")[0];
             section = Structures.SCHEMATICS_PREFIX;
-            if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(Constants.MINECOLONIES_MOD_ID, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+            //This allows mods other than minecolonies to create hut buildings and use the buildtool with them.
+            for (String origin : StructureLoadingUtils.getOriginMods().keySet())
             {
-                section = hut;
-            }
-            else
-            {
-                hut = "";
+                if (ForgeRegistries.BLOCKS.getValue(new ResourceLocation(origin, "blockhut" + hut)) != Blocks.AIR || HUTS.contains(hut))
+                {
+                    section = hut;
+                    break;
+                }
+                else
+                {
+                    hut = "";
+                }
```

Comment:
If this is meant to work for other mods, then we should change this mechanism.
Use some form of marker interface instead of using a `blockhut` prefix. or a map that is named `HUT` since this is then really not clear.

Reply:
@ShadowProtocol is doing a bigger rework with propper metadata files which will contain the propper descriptions for the future. But that rework will take a while longer. This is meant to work for the time being as a replacement for @Justen-N addon to work propperly.

Reply:
That has nothing todo with a rework.
This just needs to be defined better.

So no. Make it a marker interface on the block and call it a day.

Reply:
This was a 'quick fix' (read: bad idea) for a straight-across companion mod ( herder compatibility) But I can see where you are coming from.

Reply:
I guess, my question regarding a marker interface, is where would we put it? In Structurize? It feels like that would force an even tighter coupling between Minecolonies and Structurize.

Reply:
I suppose we could check if the block extends the AbstractBlockHut class, if that works for you.

Reply:
By the way, at the moment, all of minecolonies hut blocks still require the blockhut prefix, which is why I did not change that portion of the check. If we remove it, we'll have to refactor the blocks from Minecolonies.